src/filelock/_api.py

```diff
@@ -143,6 +136,30 @@ def _initialize(  # noqa: PLR0913
             to pass the same object around.
 
         """
+        if is_singleton and hasattr(self, "_context"):
+            # test whether other parameters match existing instance.
+            if not self.is_singleton:
+                msg = "__init__ should only be called on initialized object if it is a singleton"
+                raise RuntimeError(msg)
+
+            params_to_check = {
+                "is_thread_local": (thread_local, self.is_thread_local()),
+                "timeout": (timeout, self.timeout),
+                "mode": (mode, self.mode),
+                "blocking": (blocking, self.blocking),
+            }
+
+            non_matching_params = {name: vals for name, vals in params_to_check.items() if vals[0] != vals[1]}
```

Comment:
What if we will do something like 
```python
        non_matching_params = {name: (passed_param, set_param) for name, (passed_param, set_param) in params_to_check.items() if passed_param != set_param}
        
        if not non_matching_params:
            return
```

It will add a little bit more readable (better than `vals[0] != vals[1]`.

About `if not non_matching_params`:
1.  Using `not non_matching_params` is more straightforward and idiomatic.
2. `not non_matching_params` avoids an explicit function call to len(), making it marginally faster.

Reply:
I don't think we should worry hear about performance.

Reply:
Sure, I agree with these changes.